### PR TITLE
support customize repository prefix of image through environment: ROOK_OPERATOR_REPO_PREFIX=myrepo.io/rook

### PR DESCRIPTION
**Pull request by weekface**
rook-operator does not support customize repository prefix of image `quay.io/rook/rookd:TAG`. 
This prevent us from using our own image repository to deploy rook-operator.

### CODE REVIEW

File: pkg/operator/api/api.go [hunk 1]
```diff
@@ -161,3 +161,7 @@ func getLabels(clusterName string) map[string]string {
 		k8sutil.ClusterAttr: clusterName,
 	}
 }
+
+func MakeRookOperatorImage(version string) string {
```
Comment (travisn) [inline pkg/operator/api/api.go]: could you put this in the k8sutil package? Then we'll have both these methods for rookd and rook-operator next to each other

File: pkg/operator/k8sutil/pod.go [hunk 2]
```diff
@@ -34,8 +35,19 @@ const (
 	PodIPEnvVar = "ROOKD_PRIVATE_IPV4"
 )
 
+func RepoPrefix() string {
+	defaultRepoPrefix := "quay.io/rook"
```
Comment (travisn) [inline pkg/operator/k8sutil/pod.go]: could you put this in a const at the file level?

File: pkg/operator/k8sutil/pod.go [hunk 3]
```diff
@@ -34,8 +35,19 @@ const (
...
+
+	var repoPrefix string
+	if repoPrefix = os.Getenv("ROOK_OPERATOR_REPO_PREFIX"); repoPrefix == "" {
```
Comment (travisn) [inline pkg/operator/k8sutil/pod.go]: also would be good to have the env var in a const so the unit tests can reference it too

**Comment by weekface** [comment]
@travisn it is just for runtime.

I cannot squash all these commits right, so i open a new pr: #416, closing this.

**Comment by bassam** [comment]
@weekface you should be able to `git rebase -i` to squash them and the `git push -f`.  This is a good guide: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by weekface** [comment]
@bassam many thanks.

**Pull request closed without merging**